validator/src/main.rs

```diff
@@ -190,17 +193,42 @@ fn initialize_ledger_path(
                 None
             }
         })
-        .next()
-        .unwrap_or_else(|| {
-            error!(
-                "Entrypoint ({:?}) is not running the RPC service",
-                entrypoint.gossip
-            );
-            exit(1);
-        });
+        .next();
 
-    let client = RpcClient::new_socket(rpc_addr);
-    let genesis_hash = client.get_genesis_hash().map_err(|err| err.to_string())?;
+    if let Some(rpc_addr) = rpc_addr {
+        Ok((rpc_addr, RpcClient::new_socket(rpc_addr)))
+    } else {
+        Err(format!(
+            "No node including entrypoint ({:?}) is running the RPC service",
```

Comment:
A bit nicer message. We iterate over nodes in fact.

Reply:
I'm pretty sure we only talk to the entrypoint's RPC port here.  Where do you see otherwise?

Reply:
Yeah, I was wrong. I got misconception by `filter_map`. So, I changed to `find_map`: https://github.com/solana-labs/solana/pull/6790/files#diff-1f422d7e73618aca7bce9935356de012R185

According to the rust doc, these should be equivalent: https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.find_map